tests/Orckestra.Composer.Cart.Tests/Services/Order/OrderHistoryViewService_GetOrderDetailViewModelAsync.cs

```diff
@@ -127,5 +129,141 @@ public async Task WHEN_order_is_not_customer_one_SHOULD_return_null()
             //Assert
             result.Should().BeNull();
         }
+
+        [Test]
+        [TestCase("New|Pending", new string[] { "New" })]
+        [TestCase("New|Pending", new string[] { "Pending" })]
+        [TestCase("New|Pending", new string[] { "New", "Pending" })]
+        public async Task WHEN_valid_request_EditableShipmentStates_SHOULD_return_TRUE(string editableShipmentStates, string[] cartShipmentStatuses)
+        {
+            //Arrange
+            var customerId = Guid.NewGuid();
+
+            var orderHistoryViewModelService = _container.CreateInstance<OrderHistoryViewService>();
+            _container.GetMock<IOrderRepository>()
+                .Setup(r => r.GetOrderAsync(It.IsAny<GetCustomerOrderParam>()))
+                .ReturnsAsync(new Overture.ServiceModel.Orders.Order
+                {
+                    CustomerId = customerId.ToString()
+                });
+            
+            _container.GetMock<IOrderRepository>()
+                .Setup(r => r.GetOrderSettings(It.IsAny<string>()))
+                .ReturnsAsync(new OrderSettings()
+                {
+                    EditableShipmentStates = editableShipmentStates
+
+                });
+
+            var shipmentList = new List<Shipment>();
+            cartShipmentStatuses.ToList().ForEach(
+                item => shipmentList.Add(new Shipment(){Status = item}));
+            
+            _container.GetMock<IOrderRepository>()
+                .Setup(r => r.GetOrderAsync(It.IsAny<GetOrderParam>()))
+                .ReturnsAsync(new Overture.ServiceModel.Orders.Order()
+                {
+                    Cart = new Overture.ServiceModel.Orders.Cart {
+                        Shipments = shipmentList,
+                        Total = 3
+                    },
+                    CustomerId = customerId.ToString()
+                });
+            
+            _container.GetMock<IOrderDetailsViewModelFactory>()
+                .Setup(r => r.CreateViewModel(It.IsAny<CreateOrderDetailViewModelParam>()))
+                .Returns(new OrderDetailViewModel()
+                {
+                    OrderInfos = new OrderDetailInfoViewModel()
+                    {
+                        IsOrderEditable = true
+                    }
+                });
```

Comment:
before result.OrderInfos.IsOrderEditable.Should().BeTrue(); you set up here true.